hamilton/driver.py

```diff
@@ -260,6 +262,23 @@ class Driver:
         dr = driver.Driver(config, module, adapter=adapter)
     """
 
+    def __getstate__(self):
+        # Copy the object's state from self.__dict__
+        state = self.__dict__.copy()
+        # Remove the unpicklable entries -- right now it's the modules tracked.
+        state["graph_module_names"] = [
+            importlib.util.find_spec(m.__name__).name for m in state["graph_modules"]
+        ]
+        del state["graph_modules"]  # remove from state
+        return state
+
+    def __setstate__(self, state):
+        # Restore instance attributes
+        self.__dict__.update(state)
+        # Reinitialize the unpicklable entries
+        # assumption is that the modules are importable in the new process
+        self.graph_modules = [importlib.import_module(n) for n in state["graph_module_names"]]
```

Comment:
We should add a try/catch with an error and instructions on how to do this. Error in `__setstate__` can ber confusing.